consensus/hotstuff/cruisectl/block_time_controller_test.go

```diff
@@ -232,50 +232,78 @@ func (bs *BlockTimeControllerSuite) TestInit_EpochFallbackTriggered() {
 	bs.AssertCorrectInitialization()
 }
 
-// TestEpochFallbackTriggered tests epoch fallback:
-//   - the GetProposalTiming should revert to default
-//   - duplicate events should be no-ops
-func (bs *BlockTimeControllerSuite) TestEpochFallbackTriggered() {
+// TestOnEpochExtended ensures that the epoch configuration is updated when EpochExtended events are processed.
+func (bs *BlockTimeControllerSuite) TestOnEpochExtended() {
+	nextEpoch := mockprotocol.NewEpoch(bs.T())
+	nextEpoch.On("Counter").Return(bs.epochCounter+1, nil)
+	nextEpoch.On("FinalView").Return(bs.curEpochFinalView*2, nil)
+	nextEpoch.On("TargetDuration").Return(bs.EpochDurationSeconds(), nil)
+	nextEpoch.On("TargetEndTime").Return(bs.curEpochTargetEndTime+bs.EpochDurationSeconds(), nil)
+	bs.epochs.Add(nextEpoch)
```

Comment:
What's the reasoning for adding a `nextEpoch` here?

Reply:
Good catch https://github.com/onflow/flow-go/pull/6102/commits/f2d65065d7f9883aca707699d96be8bd8a3a9cc3